File: SYP.Web/Modules/Warehouse/StockEntryDetails/StockEntryDetailsColumns.cs
namespace SYP.Warehouse.Columns;

[ColumnsScript("Warehouse.StockEntryDetails")]
[BasedOnRow(typeof(StockEntryDetailsRow), CheckNames = true)]
public class StockEntryDetailsColumns
{
    [EditLink, Width(120)]
    public string ProductCode { get; set; }

    [Width(250)]
    public string ProductName { get; set; }

    [Width(120), AlignRight]
    public decimal Quantity { get; set; }

    [Width(200)]
    public string Notes { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Warehouse/StockEntryDetailsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { StockEntryDetailsRow } from "./StockEntryDetailsRow";

export interface StockEntryDetailsColumns {
    ProductCode: Column<StockEntryDetailsRow>;
    ProductName: Column<StockEntryDetailsRow>;
    Quantity: Column<StockEntryDetailsRow>;
    Currency: Column<StockEntryDetailsRow>;
    UnitPrice: Column<StockEntryDetailsRow>;
    Notes: Column<StockEntryDetailsRow>;
}

export class StockEntryDetailsColumns extends ColumnsBase<StockEntryDetailsRow> {
    static readonly columnsKey = 'Warehouse.StockEntryDetails';
    static readonly Fields = fieldsProxy<StockEntryDetailsColumns>();
}